gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/resources/webroot/views/mfa_challenge.html

```diff
@@ -59,7 +63,9 @@
                             </span>
                     </div>
                     <input type="hidden" th:name="${_csrf.parameterName}" th:value="${_csrf.token}"/>
-                    <input type="hidden" th:name="factorId" th:value="${factor.id}"/>
+                    <input id="factorId" type="hidden" th:name="factorId" th:value="${factor.id}"/>
+                    <input th:if="${factor.factorType.type == 'FIDO2'}" type="hidden" id="webAuthnLoginPath" th:value="${webAuthnLoginPath}"/>
+                    <input th:if="${factor.factorType.type == 'FIDO2'}" type="hidden" id="userName" th:value="${userName}"/>
```

Comment:
Do we need this ?

Reply:
Yes

line 458: userName is required as in the WebAuthnLoginEndpoint#getCredentials method line 124 is ctx.getBodyAsJson(). This line receives userName in regular passwordless login process as user provides the user name. For Fido2 challenge user is not providing the user name manually, hence we need this.
line 457: used in the webauthn-fido2-login.js